upgrade/upgrade.go

```diff
@@ -249,3 +253,38 @@ func doResourceUpgrade(namespace string, lhClient *lhclientset.Clientset, kubeCl
 
 	return upgradeutil.CreateOrUpdateLonghornVersionSetting(namespace, lhClient)
 }
+
+func waitForOldLonghornManagersToBeFullyRemoved(namespace, managerImage string, kubeClient *clientset.Clientset) error {
+	logrus.Info("Waiting for for old Longhorn manager pods to be fully removed")
+	for i := 0; i < 600; i++ {
+		managerPods, err := upgradeutil.ListManagerPods(namespace, kubeClient)
+		if err != nil {
+			return err
+		}
+		foundOldManager := false
+		for _, pod := range managerPods {
+			if isOldManagerPod(pod, managerImage) {
+				foundOldManager = true
+				break
+			}
+		}
+		if !foundOldManager {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	return fmt.Errorf("timed out while waitForOldLonghornManagersToBeFullyRemoved")
+}
+
+func isOldManagerPod(pod corev1.Pod, managerImage string) bool {
+	for _, container := range pod.Spec.Containers {
+		if container.Name == "longhorn-manager" {
+			if container.Image != managerImage {
+				logrus.Infof("Found old longhorn manager: %v with image %v", pod.Name, container.Image)
+				return true
```

Comment:
nit: `isOldManagerPod` is a query function (ex: getter/is function), to prevent verbose log when calling the function each time, it would be better to have the caller to determine if it wants to have a log. This is not a strict rule, but it's a general convention to control how/when log should be used. 

Or you can change this to `checkOldManagerPodExist`, then return error which including the error log, then return back to the caller to determine if it wants to have a log or not.

Reply:
Good point! Thank you. Fixed